### PR TITLE
feat: add the empty route test and empty map test

### PR DESCRIPTION
**Pull request by HansOersted**
## Description

Add the empty route test and empty map test.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_lanelet2_extension/lib/route_checker.cpp [hunk 1]
```diff
@@ -26,6 +26,10 @@ namespace lanelet::utils
 bool route::isRouteValid(
   const LaneletRoute & route_msg, const lanelet::LaneletMapPtr lanelet_map_ptr_)
 {
+  if (route_msg.segments.empty()) {
+    std::cerr << "Route is empty, returning false." << std::endl;
+    return false;
+  }
```
Comment (youtalk) [inline autoware_lanelet2_extension/lib/route_checker.cpp:32]: Whether an empty route segment is considered invalid is a matter of interpretation.
